transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java

```diff
@@ -194,8 +194,7 @@ private void decrementPendingOutboundBytes(long size, boolean invokeLater, boole
         }
 
         long newWriteBufferSize = TOTAL_PENDING_SIZE_UPDATER.addAndGet(this, -size);
-        if (notifyWritability && (newWriteBufferSize == 0
-            || newWriteBufferSize <= channel.config().getWriteBufferLowWaterMark())) {
+        if (notifyWritability && newWriteBufferSize < channel.config().getWriteBufferLowWaterMark()) {
```

Comment:
@normanmaurer 

Yes. This only affects the case where the low watermark is 0 and the pending bytes are 0. The [https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/WriteBufferWaterMark.java#L56](WriteBufferWaterMark) API does allow for the low and high watermarks to be `0`.

Is there a specific reason why we need to handle the zero case differently?